packages/rum/src/rumSession.ts

```diff
@@ -15,19 +15,53 @@ export enum RumSessionType {
   TRACKED_WITHOUT_RESOURCES = '2',
 }
 
+class StoredSession {
```

Comment:
What's the purpose of the `StoredSession`? I don't get it 🤔

Reply:
I agree this is non trivial and definitely should have write more doc on the subject. Currently, the most straightfoward way to get the session id is reading it directly from the cookie (there is a cache, but it doesn't help with the issue).

In RUM, we want to end the current view and start a new one when the session is renewed (= a new session is created when the previous session is expired). To end the current view, we need to send a final view event to the intake, to update the "duration" (and potentialy other metrics). Of course we need to send this event with the id of the ending session, but since the cookie may already be changed (ex: from another tab), we can't be sure we still have this id.

This is why we previously stored the current session id in the `viewContext` variable, to be sure to have a consistent id while the view is alive.

I wanted to move this storage elsewhere, because its weird that the `viewCollection` logic is in charge of the session id. This is where the `StoredSession` class takes place: an object to access the session id, updated only in a controled way.

Reply:
This change have been reverted for now, let's consider this conversation resolved :p